api/src/main/java/io/opencensus/trace/export/SampledSpanStore.java

```diff
@@ -132,6 +132,13 @@ static SampledSpanStore newNoopSampledSpanStore() {
   @PublicForTesting
   public abstract Set<String> getRegisteredSpanNamesForCollection();
 
+  /**
+   * This forces any underlying event queue to flush any pending events and shutdown and handlers.
+   *
+   * @since 0.13
+   */
+  public abstract void shutdown();
```

Comment:
I am not so sure if we should expose this API in `RunningSpanStore`.
For me, adding a `shutdown()` in `ExportComponent` is fine - that could be the entry for users and it hides all the details.

Reply:
What would a shutdown method do in RunningSpanStore?  I don't think it has a separate thread.

Reply:
Maybe make it protected?

Reply:
@sebright sorry, I meant `SampledSpanStore` in the first place. But for `RunningSpanStore`, a `shutdown` does make sense. For a `RunningSpanStore`, one might want to close, set proper status and sample option and export some important running spans before the program exit unexpectedly - I made that up :) 

Back to the topic, what I originally meant was that we should probably only expose `ExportComponent.shutdown()`. Other subsequent `shutdown` methods should be hide from users (by making them protected, or mark as `Internal`).

Reply:
Yes, I like that idea.  Users probably wouldn't need to call shutdown on only one part of the TraceComponent.  I think that the new methods on SampledSpanStore and SpanExporter are only called from the implementation anyway, so they could be completely removed from the superclasses under api/.

Reply:
i’ll wait on guidance after you have review the new commit.

Reply:
I would prefer to avoid exposing new APIs until they are needed.  Did you only add the methods so that they could be called by `ExportComponent.shutdown`, or is there a use case for only calling `shutdown` or `flush` on a subset of the components?